### ISSUE TITLE
What should I do to use the Soft max 

### ISSUE THREAD

**Issue by Oussamab21** [AaronX121/Soft-Decision-Tree#3]
Hi what should I do to update the leaf nodes to use a Soft Max instead of being nn.Linear ?

Thank you

**Comment by xuyxu** [comment]
If you mean that you want each leaf node to predict exactly one probability distribution, I think you should store predictions of leaf nodes using nn.Parameterlist() instead of nn.Linear. Furthermore, codes on the forward process should be slightly different (you may need to further use nn.functional.softmax()).

After thinking for a while, I think this job can still be done with some efforts. Good luck !
